src/components/ItemList.js

```diff
@@ -1,5 +1,43 @@
+import { collection, query, where, onSnapshot } from 'firebase/firestore';
+import { useEffect, useState } from 'react';
+import { db } from '../lib/firebase.js';
+import { Link } from 'react-router-dom';
+
 const ItemList = () => {
-  return <h1>Item List</h1>;
+  const [docs, setDocs] = useState([]);
+
+  useEffect(() => {
+    const token = localStorage.getItem('token');
+    const q = query(
+      collection(db, 'shopping-list'),
+      where('token', '==', token),
```

Comment:
This is cool! I don't recall this filtering capability on the Firebase API! Nice to not have to consume the entire list locally and filter!